examples/mutual-auth-alb/main.tf

```diff
@@ -0,0 +1,301 @@
+        port                = "traffic-port"
+        healthy_threshold   = 3
+        unhealthy_threshold = 3
+        timeout             = 6
+        protocol            = "HTTP"
+        matcher             = "200-399"
+      }
+
+      protocol_version = "HTTP1"
+      target_id        = aws_instance.this.id
+      port             = 80
+      tags = {
+        InstanceTargetGroupTag = "baz"
+      }
+    }
+  }
+
+  # Route53 Record(s)
+  route53_records = {
+    A = {
+      name    = local.name
+      type    = "A"
+      zone_id = data.aws_route53_zone.this.id
+    }
+    AAAA = {
+      name    = local.name
+      type    = "AAAA"
+      zone_id = data.aws_route53_zone.this.id
+    }
+  }
+}
+
+module "trust_store" {
+  source = "../../modules/lb_trust_store"
+
+  name                             = "${local.name}-trust-store"
+  ca_certificates_bundle_s3_bucket = module.certificate_bucket.s3_bucket_id
+  ca_certificates_bundle_s3_key    = "ca_cert/RootCA.pem"
+  create_trust_store_revocation    = true
+  revocation_lists = {
+    crl_1 = {
+      revocations_s3_bucket = module.certificate_bucket.s3_bucket_id
+      revocations_s3_key    = "crl/crl.pem"
+    }
+  }
+
+  depends_on = [
+    module.ca_cert_object,
+    module.crl_object,
+  ]
+}
+
+################################################################################
+# Supporting resources
+################################################################################
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 5.0"
+
+  name = local.name
+  cidr = local.vpc_cidr
+
+  azs             = local.azs
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 4, k)]
+  public_subnets  = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k + 48)]
+
+  tags = local.tags
+}
+
+data "aws_ssm_parameter" "al2" {
+  name = "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2"
+}
+
+resource "aws_instance" "this" {
+  ami           = data.aws_ssm_parameter.al2.value
+  instance_type = "t3.nano"
+  subnet_id     = element(module.vpc.private_subnets, 0)
+}
+
+module "log_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "~> 3.0"
+
+  bucket_prefix = "${local.name}-logs-"
+  acl           = "log-delivery-write"
+
+  # For example only
+  force_destroy = true
+
+  control_object_ownership = true
+  object_ownership         = "ObjectWriter"
+
+  attach_elb_log_delivery_policy = true # Required for ALB logs
+  attach_lb_log_delivery_policy  = true # Required for ALB/NLB logs
+
+  attach_deny_insecure_transport_policy = true
+  attach_require_latest_tls_policy      = true
+
+  tags = local.tags
+}
+
+data "aws_route53_zone" "this" {
+  name = var.domain_name
+}
+
+module "acm" {
+  source  = "terraform-aws-modules/acm/aws"
+  version = "~> 4.0"
+
+  domain_name = "*.${var.domain_name}"
+  zone_id     = data.aws_route53_zone.this.id
+}
+
+resource "null_resource" "generate_certificates" {
```

Comment:
Can we use a [TLS provider](https://registry.terraform.io/providers/hashicorp/tls/latest) to make certificates instead of shell scripts?